staging/src/k8s.io/client-go/tools/cache/shared_informer_test.go

```diff
@@ -357,3 +358,45 @@ func TestSharedInformerErrorHandling(t *testing.T) {
 	}
 	close(stop)
 }
+
+// TestSharedInformerRunWithStopOptions runs an informer with StopOnListError
+// set to always return true. It tests that when the underlying reflector does reach a
+// list error (upon trying to add an object) that the informer stopps running before a
+// given timout.
+func TestSharedInformerRunWithStopOptions(t *testing.T) {
+	table := []struct {
+		stopOnListError bool
+	}{
+		{true},
+		{false},
+	}
+	for _, item := range table {
+		source := fcache.NewFakeControllerSource()
+		source.Add(&v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "pod1"}})
+		source.ListError = fmt.Errorf("Access Denied")
+
+		informer := NewSharedInformer(source, &v1.Pod{}, 1*time.Second).(*sharedIndexInformer)
+
+		ctx, cancel := context.WithCancel(context.Background())
+		go func() {
+			// confirm the informer stops running when it hits a list error
+			defer cancel()
+			informer.RunWithStopOptions(ctx, StopOptions{
+				StopOnListError: func(err error) bool {
+					return item.stopOnListError
+				},
+			})
+		}()
+
+		select {
+		case <-ctx.Done():
+			if !item.stopOnListError {
+				t.Errorf("shared informer should NOT have stopped when stopOnListError is false")
+			}
+		case <-time.After(time.Second):
```

Comment:
what about using ForeverTestTimeout (30s)? or you believe 1s is sufficiently long?

Reply:
Same as my comment in `reflector_test.go`, it actually must wait this long in order to validate the stopOnListError=false case, so I've bumped it down to 200ms.